rpc/src/fetch.rs

```diff
@@ -29,7 +28,27 @@ pub async fn proxy_query_to_history_subnet(
     match resp_rx.recv().await {
         Some(val) => match val {
             Ok(result) => Ok(result),
-            Err(msg) => Err(RpcServeError::Message(msg)),
+            Err(msg) => {
```

Comment:
I think, we also need to update this for [state](https://github.com/ethereum/trin/blob/ebc372affd7bc13d254f955d2517ad7adde1fdd8/rpc/src/state_rpc.rs#L31) network.

Reply:
None of the endpoints that would use this kind of error are actually supported inside the [state jsonpc handler](https://github.com/ethereum/trin/blob/master/trin-state/src/jsonrpc.rs#L32-L47) so for the sake of moving this pr ahead, I'm going to leave state alone. But this is definitely something that @morph-dev should keep in mind as he develops the state network's capabilities

Reply:
Yes, that's correct. I finished PR #1175 that implements those endpoints for state network, but it's easier if you merge this one first, and I integrate changes there afterwards.